superset/connectors/sqla/models.py

```diff
@@ -833,6 +833,9 @@ def get_rendered_sql(
         """
         Render sql with template engine (Jinja).
         """
+        if not self.sql:
+            return ""
```

Comment:
should this `raise QueryObjectValidationError(_("Virtual dataset query cannot be empty"))` like below instead of silently failing? seems like if we move the check here, then we can get rid of the check on line 851 too

Reply:
I think this makes sense. Will try it out and see if this works.